lib/source_file.h

```diff
@@ -126,6 +127,14 @@ class SourceInfo final {
     int line = -1;
     int column = -1;
     cstring srcBrief = ""_cs;
+
+    // add a comment to 'before' or 'after' array of comments
+    void addBefore(Util::Comment *comment) const;
+    void addAfter(Util::Comment *comment) const;
```

Comment:
I suggest naming this `addCommentsBefore`/`addCommentsAfter` to increase clarity.